include/RAJA/policy/openmp/reduce.hpp

```diff
@@ -568,499 +235,49 @@ class ReduceMaxLoc<omp_reduce, T>
-  {
-    int tid = omp_get_thread_num();
-    Reduce()(*asT(tid), *asIndex(tid), val, idx);
-    return *this;
-  }
-
-private:
-
-  static constexpr const int s_block_offset =
-      COHERENCE_BLOCK_SIZE / sizeof(CPUReductionBlockDataType);
-  static constexpr const int s_idx_offset = COHERENCE_BLOCK_SIZE / sizeof(Index_type);
-
-  int m_myID;
-  CPUReductionBlockDataType *m_blockdata;
-  Index_type *m_idxdata;
-  T m_reduced_val;
-  Index_type m_reduced_idx;
-  bool m_is_copy;
+  using Base = detail::BaseReduceMinLoc<T, detail::ReduceOMPOrdered>;
+  using Base::Base;
 };
 
-/*!
- ******************************************************************************
- *
- * \brief  Sum reducer class template for use in OpenMP execution.
- *
- *         For usage example, see reducers.hxx.
- *
- ******************************************************************************
- */
 template <typename T>
-class ReduceSum<omp_reduce_ordered, T>
+class ReduceMaxLoc<omp_reduce_ordered, T>
+    : public detail::BaseReduceMaxLoc<T, detail::ReduceOMPOrdered>
 {
 public:
-
-  //! Default ctor is declared private and not implemented.
-  ReduceSum() = delete;
-
-  //! Constructor takes default value (default ctor is disabled).
-  explicit ReduceSum(T init_val)
-    : m_myID(getCPUReductionId()),
-      m_blockdata(getCPUReductionMemBlock(m_myID)),
-      m_init_val(init_val),
-      m_reduced_val(0),
-      m_is_copy(false)
-  {
-    int nthreads = omp_get_max_threads();
-#pragma omp parallel for schedule(static, 1)
-    for (int i = 0; i < nthreads; ++i) {
-      m_blockdata[i * s_block_offset] = 0;
-    }
-  }
-
-  //
-  // Copy ctor.
-  //
-  ReduceSum(const ReduceSum &other)
-    : m_myID(other.m_myID),
-      m_blockdata(other.m_blockdata),
-      m_init_val(other.m_init_val),
-      m_reduced_val(other.m_reduced_val),
-      m_is_copy(true)
-  {
-  }
-
-  //! Destructor releases shared memory block and reduction id
-  ~ReduceSum()
-  {
-    if (m_is_copy)
-      return;
-    releaseCPUReductionId(m_myID);
-  }
-
-  //! Operator that returns reduced sum value.
-  operator T()
-  {
-    T tmp_reduced_val = static_cast<T>(0);
-    int nthreads = omp_get_max_threads();
-    for (int i = 0; i < nthreads; ++i) {
-      tmp_reduced_val += static_cast<T>(m_blockdata[i * s_block_offset]);
-    }
-    m_reduced_val = m_init_val + tmp_reduced_val;
-
-    return m_reduced_val;
-  }
-
-  //! Method that returns sum value.
-  T get() { return operator T(); }
-
-  //! += operator that adds value to sum for current thread.
-  ReduceSum operator+=(T val) const
-  {
-    int tid = omp_get_thread_num();
-    m_blockdata[tid * s_block_offset] += val;
-    return *this;
-  }
-
-private:
-
-  static const int s_block_offset =
-      COHERENCE_BLOCK_SIZE / sizeof(CPUReductionBlockDataType);
-
-  int m_myID;
-  CPUReductionBlockDataType *m_blockdata;
-  T m_init_val;
-  T m_reduced_val;
-  bool m_is_copy;
-
+  using Base = detail::BaseReduceMaxLoc<T, detail::ReduceOMPOrdered>;
```

Comment:
Formatting?

Reply:
Ignore this comment also. Looks fine when not viewed in a narrow browser window.